command/validate.go

```diff
@@ -80,6 +83,47 @@ func (c *ValidateCommand) Run(args []string) int {
 		}
 	}
 
+	// Check if any of the configuration is fixable
+	var templateData map[string]interface{}
+	json.Unmarshal(tpl.RawContents, &templateData)
+	input := make(map[string]interface{})
+	for k, v := range templateData {
+		input[k] = v
+	}
+	for _, name := range fix.FixerOrder {
+		var err error
+		fixer, ok := fix.Fixers[name]
+		if !ok {
+			panic("fixer not found: " + name)
+		}
+		input, err = fixer.Fix(input)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error fixing: %s", err))
+			return 1
+		}
+	}
+	// delete empty top-level keys since the fixers seem to add them
+	// willy-nilly
+	for k := range input {
+		ml, ok := input[k].([]map[string]interface{})
+		if !ok {
+			continue
+		}
+		if len(ml) == 0 {
+			delete(input, k)
+		}
+	}
```

Comment:
Deleting these means that if a user has empty keys set this will always tell them they need to fix their template, won't it?

Reply:
yeah, if you have like `{"post-processors": []}` then it'll complain. Didn't seem like a likely scenario, and the diffing library I'm using has a bug that makes this not as slick as I was hoping (https://github.com/google/go-cmp/issues/73)

Will think about it some more. I can think of a few workarounds tho they might be a bit hacky